### ISSUE TITLE
priority about showNavigationBar={false} and hideNavBar={true}

### ISSUE THREAD

**Issue by zxcpoiu** [aksonov/react-native-router-flux#262]
Those two props should be the same.

https://github.com/aksonov/react-native-router-flux/issues/255#issuecomment-189457811
https://github.com/aksonov/react-native-router-flux/issues/255#issuecomment-189458270

But even though I have always set `hideNavBar={true}`, I still encounter issue #258 which can avoid by set `showNavigationBar={false}` as a workaround if you don't need the nav bar.

I guess:
* `hideNavBar` is a prop in `react-native-router-flux`
* `showNavigationBar` is a prop in `ex-navigator`

#### should we prioritize/merge them?

for example:

* priority of `hideNavBar` is higher than `showNavigationBar` and overrides it.
* setting in `Router` is higher than `Route` and overrides it.
* a function in `ExRouter.js` handle those prop and return `true/false`

and we can always set prop `showNavigationBar` specifically in `<ExNavigator>` in `ExRouter.js`

below is a similar check like `_renderNavigationBar()` in `ExRouter.js`:

```
// Something like this

shoudHideNavBar(props) {
    const route = this.props.router.nextRoute || this.props.router.currentRoute;
    const hideNavBar = route.props.hideNavBar || this.props.router.props.hideNavBar || route.props.hideNavBar;
    const showNavigationBar = ........

    if (hideNavBar !== 'undefined') {
        .......
    }
    return true/false;
}

```

what do you think?

**Comment by lrettig** [comment]
Agree that this is terribly confusing, I'd love to see this cleaned up! Thanks for pointing it out

**Comment by aksonov** [comment]
What is problem to use hideNavBar only?

**Comment by zxcpoiu** [comment]
At first, there are no problems at all.

#### first reason:
When I tried to find a workaround for #258 ( fixed by 2.3.7 ), which can be avoided if you set not to render NavBar at all.

At that time, I noticed that even though I use `hideNavBar={true}`, the app was still trying to reder NavBar, so error occurs.
Only if I use `showNavigationBar={false}` specifically that I can prevent the error.

I don't know why it was, but maybe it is fixed, so it won't have any problems to me now.

#### second reason:
In [Example.js line 86 ](https://github.com/aksonov/react-native-router-flux/blob/master/Example/Example.js#L86), it use `showNavigationBar={false}` instad of `hideNavBar={true}`.

I think this is the most confusing part to newbies like me.

#### That's why I think it may be better to:

* have a wrapper function that prioritize them or hide the lower api `showNavigationBar` at all.
* always use `hideNavBar` in any Example code. 

yup, maybe add a wrapper for this is a little be redundant.
I just make a PR to use `hideNavBar` in example instead `showNavigationBar`.
#270
